cmake/RunReverseDD.cmake

```diff
@@ -0,0 +1,201 @@
+        file(COPY "${SOURCE_DIR}/idd/SlabGHT.idd" DESTINATION "${OUTPUT_DIR_PATH}")
+        # Find and run slab
+        find_program(SLAB_EXE Slab PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                                                           NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+        message("Executing Slab from ${SLAB_EXE}")
+        execute_process(COMMAND "${SLAB_EXE}" WORKING_DIRECTORY "${OUTPUT_DIR_PATH}")
+        # Then copy slab results into the expanded file
+        file(READ "${OUTPUT_DIR_PATH}/SLABSurfaceTemps.TXT" SLAB_CONTENTS)
+        file(APPEND "${OUTPUT_DIR_PATH}/expanded.idf" "${SLAB_CONTENTS}")
+      endif()
+
+      if("${BASEMENT_RESULT}" GREATER -1)
+        # Copy files needed for Basement
+        file(COPY "${SOURCE_DIR}/idd/BasementGHT.idd" DESTINATION "${OUTPUT_DIR_PATH}")
+        # Find and run basement
+        find_program(BASEMENT_EXE Basement PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                                                                   NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+        message("Executing Basement from ${BASEMENT_EXE}")
+        execute_process(COMMAND "${BASEMENT_EXE}" WORKING_DIRECTORY "${OUTPUT_DIR_PATH}")
+        # Then copy basement results into the expanded file
+        file(READ "${OUTPUT_DIR_PATH}/EPObjects.TXT" BASEMENT_CONTENTS)
+        file(APPEND "${OUTPUT_DIR_PATH}/expanded.idf" "${BASEMENT_CONTENTS}")
+      endif()
+
+      set(IDF_PATH "${OUTPUT_DIR_PATH}/expanded.idf")
+
+    endif() # expand objects found something and created expanded.idf
+
+    list(FIND ENERGYPLUS_FLAGS_LIST -x EXPAND_RESULT)
+
+    if("${EXPAND_RESULT}" GREATER -1)
+      find_program(EXPANDOBJECTS_EXE ExpandObjects PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH
+                                                                           NO_SYSTEM_ENVIRONMENT_PATH NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+      # Move to executable directory
+      execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${EXPANDOBJECTS_EXE}" "${EXE_PATH}")
+    endif()
+
+    list(FIND ENERGYPLUS_FLAGS_LIST -r READVARS_RESULT)
+
+    if("${READVARS_RESULT}" GREATER -1)
+      find_program(READVARS_EXE ReadVarsESO PATHS "${PRODUCT_PATH}" NO_DEFAULT_PATH NO_CMAKE_ENVIRONMENT_PATH NO_CMAKE_PATH NO_SYSTEM_ENVIRONMENT_PATH
+                                                                    NO_CMAKE_SYSTEM_PATH NO_CMAKE_FIND_ROOT_PATH)
+      # Move to executable directory
+      execute_process(COMMAND ${CMAKE_COMMAND} -E copy_if_different "${READVARS_EXE}" "${EXE_PATH}")
+    endif()
+
+  endif() # build fortran
+
+  if(WIN32)
+    set(ECHO_CMD cmd /C echo.)
+  else()
+    set(ECHO_CMD "echo")
+  endif()
+
+  # set the reverse design day environment variable for EnergyPlus
+  set(ENV{REVERSEDD} "${V}")
```

Comment:
This line sets the `REVERSEDD` environment variable prior to calling EnergyPlus.  When this is enabled, EnergyPlus will switch the order of the reverse design days.  There is some extra complication nowadays though, because the HVAC sizing simulation design days are also involved in the reordering.  Adds some complexity to the post processing.